### DISCUSSION TITLE
CI check to enforce whitespace consistency

### DISCUSSION THREAD

**Discussion by mtlynch** [jonaswinkler/paperless-ng#1108] [Feature Requests]
I notice that paperless-ng's source files have inconsistent whitespace. Some files end with a trailing newline, whereas others don't.

Several files also include lines that end in extraneous blank spaces:

```
$ grep "\s$" -l --binary-files=without-match --exclude="*.svg" --exclude="*.eps" ./ -R
./docs/configuration.rst
./docs/contributing.rst
./docs/usage_overview.rst
./docs/extending.rst
./docs/index.rst
...
```

Would you be open to a PR that fixes both of these issues repo-wide and introduces CI checks that prevent them from recurring? I can submit the fixes, as I've done the same thing in all of my repos:

* [check-trailing-whitespace](https://github.com/tiny-pilot/tinypilot/blob/42c3768be678e11d8bc449fd30a42a0ba501768b/dev-scripts/check-trailing-whitespace)
* [check-trailing-newline](https://github.com/tiny-pilot/tinypilot/blob/42c3768be678e11d8bc449fd30a42a0ba501768b/dev-scripts/check-trailing-newline)

If you're open to a similar implementation, where would be a good place to put the checker scripts? `scripts` doesn't seem like the right fit because these are not scripts that serve a purpose in production. Would a top-level folder called `dev-scripts` work? And I could add a Github action to call these scripts on PRs.

**Comment by jonaswinkler** [comment]
Hello!

Good idea. I'd say `.github/workflow-scripts/...` would be a good location for this.

**Reply by mtlynch** [reply]
Cool, I've submitted the PR here:

https://github.com/jonaswinkler/paperless-ng/pull/1114
